src/backend/cdb/cdbmutate.c

```diff
@@ -1009,16 +1149,19 @@ apply_motion_mutator(Node *node, ApplyMotionState *context)
 	plan->nMotionNodes = context->nextMotionID - saveNextMotionID;
 	plan->nInitPlans = hash_get_num_entries(context->planid_subplans) - saveNumInitPlans;
 
-	/*
-	 * Remember if this was a Motion node. This is used at the top of the
-	 * tree, with MOVEMENT_EXPLICIT, to avoid adding an explicit motion, if
-	 * there were no Motion in the subtree. Note that this does not take
-	 * InitPlans containing Motion nodes into account. InitPlans are executed
-	 * as a separate step before the main plan, and hence any Motion nodes in
-	 * them don't need to affect the way the main plan is executed.
-	 */
-	if (IsA(newnode, Motion))
-		context->containMotionNodes = true;
+	if (context->mt.isChecking && IsA(node, Motion))
+	{
+		Motion	   *motion = (Motion *) node;
+
+		if (motion->motionType == MOTIONTYPE_HASH ||
+			(motion->motionType == MOTIONTYPE_FIXED && motion->isBroadcast))
```

Comment:
According to this comment, Broadcast Motion is always Fixed.
https://github.com/arenadata/gpdb/blob/c097366900c7699f6e10fe5f29a763792d34af41/src/include/nodes/plannodes.h#L1298-L1307